### PR TITLE
Feature/others user page

### LINKED ISSUE #396: Other's user page
## Motivations
For the moment, if you click on a user in the left menu, you get an error. Let's create a basic page.

## Subtasks
* Create a new page on `/groups/users/:id` which uses the new service [userView](https://france-ioi.github.io/algorea-devdoc/api/#operation/userView).
* Display in header either the "`first_name` `last_name` (`login`)" if first_/last_name are given (not undefined) and not null; "`login`" otherwise.
* if not null, display the website as in https://france-ioi.github.io/algorea-designs/01.Groups_06/Personal-data
* if `free_text` is not null, display a text under the header as shown in the design https://france-ioi.github.io/algorea-designs/01.Groups_06/Personal-data

### PR DESCRIPTION
**Pull request by Iloveall**
Fixes #396

### CODE REVIEW

File: src/app/shared/routing/group-route.ts [hunk 1]
```diff
@@ -7,8 +7,13 @@ export interface GroupRoute extends ContentRoute {
 /**
  * Return a url array (`commands` array) to the given group, on the given page.
  */
-export function urlArrayForGroupRoute(route: GroupRoute, page: 'edit'|'details' = 'details'): any[] {
+export function urlArrayForGroupRoute(route: GroupRoute, page: 'edit'|'details'|'users' = 'details'): any[] {
```
Comment (smadbe) [inline src/app/shared/routing/group-route.ts:10]: The page parameter is supposed to be a group page... "users" is no a subpage of group.

File: src/app/shared/routing/group-router.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ export class GroupRouter {
    * Navigate to given group, on the path page.
    * If page is not given and we are currently on a group page, use the same page. Otherwise, default to 'details'.
    */
-  navigateTo(route: GroupRoute, page?: 'edit'|'details'): void {
+  navigateTo(route: GroupRoute, page?: 'edit'|'details'|'users'): void {
```
Comment (smadbe) [inline src/app/shared/routing/group-router.ts:18]: The same everywhere here, I don't think that's a good idea to put it as the same level as 'edit' and 'details' as it is not.
For the navigation just navigate "manually" to "[ '/', 'groups', 'users', id ]" for now, out of this router. I'll see later how/if we integrate users in this group router (if they need route as well).

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/group/services/user-view.service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    })
+  )
+);
+
+export type UserView = D.TypeOf<typeof viewUserDecoder>;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserViewService {
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/group/services/user-view.service.ts:30]: I took caption from issue description - ...new service userView.

File: src/app/modules/group/pages/group-user/group-user.component.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    <div class="bg-white" *ngIf="user.webSite">
+      <alg-section i18n-label label="Personal Information" icon="fa fa-users">
+        <ul class="alg-data-list">
+          <li class="alg-data-list-item">
+            <span class="alg-data-list-item-caption" i18n>Personal Web page</span>
+            <span class="alg-data-list-item-value">{{ user.webSite }}</span>
+          </li>
+        </ul>
+      </alg-section>
+    </div>
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/group/pages/group-user/group-user.component.html:35]: Fixed

File: src/app/core/components/left-nav-tree/left-nav-tree.component.ts [hunk 5]
```diff
@@ -88,6 +90,10 @@ export class LeftNavTreeComponent implements OnChanges {
     const routeBase = { id: node.data.element.id, path: node.data.path };
     switch (this.elementType) {
       case 'group':
+        if (node.data?.element?.type === 'User') {
+          this.groupRouter.navigateTo({ ...routeBase, contentType: 'group' }, 'users');
+          return;
+        }
         this.groupRouter.navigateTo({ ...routeBase, contentType: 'group' });
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/core/components/left-nav-tree/left-nav-tree.component.ts]: Fixed

File: src/app/modules/group/pages/group-user/group-user.component.html [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  <div *ngIf="state.isError">
+    Error
+  </div>
+
+  <ng-container *ngIf="state.isReady && state.data as user">
+
+    <div class="alg-page-header">
+      <div class="alg-page-header-title">
+        <span *ngIf="user.firstName || user.lastName; else login">
+          {{ user.firstName }} {{ user.lastName }}
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/group/pages/group-user/group-user.component.html:12]: Fixed

File: src/app/modules/group/services/user-view.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/group/services/user-view.service.ts]: Fixed

File: src/app/shared/routing/group-route.ts (hunk 1 above)
Comment (Iloveall) [reply src/app/shared/routing/group-route.ts]: Fixed

File: src/app/shared/routing/group-router.ts (hunk 2 above)
Comment (Iloveall) [reply src/app/shared/routing/group-router.ts]: Fixed

File: src/app/modules/group/services/user-view.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (smadbe) [reply src/app/modules/group/services/user-view.service.ts]: > I took caption from issue description - ...new service userView.

Ah yes ok. It was just the name of the service in the backend. No need to use these names for the FE.

File: src/app/modules/group/pages/group-user/group-user.component.html [hunk 7]
```diff
@@ -1,37 +1,28 @@
...
+
+  <p *ngIf="state.isError" i18n>Error while loading the user info</p>
 
   <ng-container *ngIf="state.isReady && state.data as user">
 
     <div class="alg-page-header">
       <div class="alg-page-header-title">
-        <span *ngIf="user.firstName || user.lastName; else login">
+        <span *ngIf="user.firstName && user.lastName; else login">
           {{ user.firstName }} {{ user.lastName }}
```
Comment (smadbe) [inline src/app/modules/group/pages/group-user/group-user.component.html:12]: Still not. It is supposed to show  "first_name last_name (login)" (so the login between brackets) when  first_name/last_name are known.

**Review by smadbe — CHANGES_REQUESTED** [review]
Title still in a wrong format.

File: src/app/modules/group/pages/group-user/group-user.component.html (hunk 7 above)
Comment (Iloveall) [reply src/app/modules/group/pages/group-user/group-user.component.html]: Fixed

**Review by smadbe — APPROVED** [review]
👍

**Pull request merged**